api/adaptive_load/metric_spec.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/scoring_function.proto";
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// Identifies a feedback metric.
+message MetricSpec {
+  // Name of the metric to evaluate. For the set of built-in metric names, see
+  // source/adaptive_load/metrics_plugin_impl.cc.
+  string metric_name = 1 [(validate.rules).string.min_len = 1];
+  // Name of the MetricsPlugin providing the metric ("builtin" for built-in).
+  string metrics_plugin_name = 2 [(validate.rules).string.min_len = 1];
+}
+
+// Specifies a simple upper or lower threshold, or configures a scoring
+// function.
+message ThresholdSpec {
+  // A threshold for evaluating a metric value.
+  oneof threshold_spec {
+    // Simple upper threshold:
+    //   metric <= upper_threshold -> WITHIN_THRESHOLD
+    //   metric > upper_threshold -> OUTSIDE_THRESHOLD
+    google.protobuf.DoubleValue upper_threshold = 1;
```

Comment:
do we need the wrappers for the double valued fields here?

Reply:
The reason for wrapping these is to be able to check which of the oneof options is set. Theoretically the default double 0.0 could be the threshold, and a threshold could be positive or negative, so in general can't inspect the value to tell if the field was set. At least not as far as I know with proto3.

Reply:
I see, makes sense